### DISCUSSION TITLE
Allow setting paths to executable for fd and bat

### DISCUSSION THREAD

**Discussion by wiget** [PatrickF1/fzf.fish#139] [Archive] [answered]
On Ubuntu `fd` is available in package `fdfind` and executable is named `fdfind` (to not conflict with `fd` from `fdclone` package). Similar for `bat` which is in `bat` package but executable is named `batcat`(to not conflict with `bat` from `bacula-console-qt`).

Having option to set executable for this two tools will be useful to make this excellent plugin working correctly on Ubuntu.

**Comment by PatrickF1** [comment, marked as answer]
Hi Artur! Thanks for the idea. Initially, a lot of Linux people would open up bug reports on fd not being available but then after I covered this in the documentation, the reports completely went away, so I am not too worried about this anymore. However, I am open to ideas to providing better support on Ubuntu. Can you elaborate on "Having option to set executable for this two tools"?

**Reply by wiget** [reply]
Maybe introduce two new variables:
- `set fzf_preview_file_cmd batcat`
- `set fzf_find_files_cmd fdfind`

**Reply by PatrickF1** [reply]
Thanks for the idea. Introducing two new variables seems overkill as they will always be named bat/fd, or batcat/fdbat. It's not something that needs to be customized. Furthermore, wouldn't the user want to just properly fix the executable names anyway? If they're going to go through all the work of setting variables for these executables, they might as well just fix it so they can use it day-to-day usage.

Unless you are saying that there are people want to use both fdfind and fdclone, or bat and bacula-console-qt, in which case that's a different problem I'm going to defer to others to fix.

What do you think?

**Reply by wiget** [reply]
I understand your point. All I wanted to say is that out of the box experience for Ubuntu (and probably Debian) users is poor. I know how to solve this for myself, so it not a problem for me.

Thanks for your time.

**Reply by PatrickF1** [reply]
> All I wanted to say is that out of the box experience for Ubuntu (and probably Debian) users is poor

I think this affects all Linux users....right? I agree this is poor onboarding experience, but I just don't know if this is up to the plugin to solve. 
However, I do want to ameliorate this with better help on this. I'm going to write better help for this in the Wiki and readme. But I actually don't know the best way to fix it. What commands did you run to fix this? I'm guessing the best way is to symlink fdcat to fd, batcat to bat?

**Reply by IlanCosman** [reply]
This certainly does not effect all Linux users. Arch for example has bat and fd as, well, bat and fd.

**Reply by PatrickF1** [reply]
Thanks for clarifying that, Ilan! Can you help out a non-Linux expert here? If you had this problem, how would you solve it? Symlink the binary to the desired name?

**Reply by PatrickF1** [reply]
Nvm I read the instructions on both projects' GitHub readmes (sorry shoulda done my research first before asking) and I see my guess was correct. I'll update the Wiki but I won't be implementing this. Thanks for understanding and thank you for your suggestion!
